pkg/services/ngalert/eval/eval.go

```diff
@@ -146,12 +151,24 @@ func GetExprRequest(ctx AlertExecCtx, data []models.AlertQuery, now time.Time) (
 			return nil, fmt.Errorf("failed to retrieve maxDatapoints from the model: %w", err)
 		}
 
+		ds, ok := datasources[q.DatasourceUID]
+		if !ok {
+			if expr.IsDataSource(q.DatasourceUID) {
+				ds = expr.DataSourceModel()
+			} else {
+				ds, err = dsCacheService.GetDatasourceByUID(q.DatasourceUID, &m.SignedInUser{OrgId: ctx.OrgID}, true)
```

Comment:
Think admin role is needed. Agree, it feels a bit weird ngalert needing to do this. #41653 introduces a new query service that might help abstracting away things for ngalert?

```suggestion
				ds, err = dsCacheService.GetDatasourceByUID(q.DatasourceUID, &m.SignedInUser{
					OrgId: ctx.OrgID,
					OrgRole: m.ROLE_ADMIN,
				}, true)
```